firmware/library/L1_Drivers/dac.hpp

```diff
@@ -1,19 +1,209 @@
+    820, 828, 836, 844, 852, 859, 867, 875,
+    883, 891, 898, 906, 914, 922, 930, 938,
+    945, 953, 961, 969, 977, 984, 992, 1000,
+    1008, 1016, 1023, 1031, 1039, 1047, 1055, 1063,
+    1070, 1078, 1086, 1094, 1102, 1109, 1117, 1125,
+    1133, 1141, 1148, 1156, 1164, 1172, 1180, 1188,
+    1195, 1203, 1211, 1219, 1227, 1234, 1242, 1250,
+    1258, 1266, 1273, 1281, 1289, 1297, 1305, 1313,
+    1320, 1328, 1336, 1344, 1352, 1359, 1367, 1375,
+    1383, 1391, 1398, 1406, 1414, 1422, 1430, 1438,
+    1445, 1453, 1461, 1469, 1477, 1484, 1492, 1500,
+    1508, 1516, 1523, 1531, 1539, 1547, 1555, 1563,
+    1570, 1578, 1586, 1594, 1602, 1609, 1617, 1625,
+    1633, 1641, 1648, 1656, 1664, 1672, 1680, 1688,
+    1695, 1703, 1711, 1719, 1727, 1734, 1742, 1750,
+    1758, 1766, 1773, 1781, 1789, 1797, 1805, 1813,
+    1820, 1828, 1836, 1844, 1852, 1859, 1867, 1875,
+    1883, 1891, 1898, 1906, 1914, 1922, 1930, 1938,
+    1945, 1953, 1961, 1969, 1977, 1984, 1992, 0
+    };
+    const uint16_t kSineLookup[256] =
+    {
+        29, 5, 1, 7, 4, 0, 6, 2, 8, 4, 30, 36, 42, 48, 54, 60, 65,
+    71, 77, 82, 88, 93, 98, 3, 8, 13, 18, 23, 28, 33, 37,
+    41, 46, 50, 54, 58, 62, 65, 69, 72, 76, 79, 82, 85, 87,
+    90, 93, 95, 97, 99, 1, 3, 4, 6, 7, 8, 9, 10, 11,
+    11, 12, 12, 12, 12, 12, 11, 11, 10, 9, 8, 7, 6, 4,
+    3, 1, 99, 97, 95, 93, 90, 87, 85, 82, 79, 76, 72, 69,
+    65, 62, 58, 54, 50, 46, 41, 37, 33, 28, 23, 18, 13, 408,
+    403, 398, 393, 388, 382, 377, 371, 365, 360, 354, 348,
+    342, 336, 330, 324, 318, 312, 306, 300, 294, 287, 281,
+    275, 269, 262, 256, 250, 243, 237, 231, 225, 218, 212,
+    206, 200, 194, 188, 182, 176, 170, 164, 158, 152, 147, 141,
+    135, 130, 124, 119, 114, 109, 104, 99, 94, 89, 84,
+    79, 75, 71, 66, 62, 58, 54, 50, 47, 43, 40, 36, 33, 30, 27,
+    25, 22, 19, 17, 15, 13, 11, 9, 8, 6, 5, 4, 3, 2, 1, 1, 0, 0, 0,
+    0, 0, 1, 1, 2, 3, 4, 5, 6, 8, 9, 11, 13, 15, 17, 19, 22, 25,
+    27, 30, 33, 36, 40, 43, 47, 50, 54, 58, 62, 66, 71, 75, 79,
+    84, 89, 94, 99, 104, 109, 114, 119, 124, 130,
+    135, 141, 147, 152, 158, 164, 170, 176, 182, 188, 194,
+    200, 206, 212, 218, 225, 231, 237, 243, 250, 255
+    };
+    // PinConfigure led3(PinConfigure::CreatePinConfigure<1, 15>());
+    /* Initialize Dac by setting the clock divider and enabling 
+      The Dac Pin withiin IOCON.  Then set the initial Bias level to 0 */
+    void Initialize(void) override
+    {
+        // Set Clock Divider to 1
+        LPC_SC->PCLKSEL = (1 << kClockDividerReg);
+        // Enable DAC Pin P0.26
+        LPC_IOCON->P0_26 = (1 << kEnableDacReg);
+
+        EnableDAC(true);
+        // Set Update Rate to 1MHz
+        LPC_DAC->CR &= ~(1 << kBiasReg);
+        // return true;
+    }
+    // The Dac ouptut is a 10 bit input and thus it is necessary to
+    // ensure dac_output is less than 1024(largest 10-bit number)
+    bool WriteDac(uint16_t dac_output) override
+    {
+        if ( dac_output < 1024 )
```

Comment:
Rather than doing this, use an SJ2_ASSERT_FATAL() for this. See i2c.hpp for an example.